openml-lightgbm/lightgbm-provider/src/test/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModelTest.java

```diff
@@ -236,4 +249,122 @@ private boolean compareDoubles(final double a, final double b, final int exclude
+     * @throws IOException          In case the process errors.
+     * @throws InterruptedException In case awaiting for the process to finish fails.
+     * @since 1.0.19
+     */
+    private boolean compareModelFilesAndDoPrettyReport(
+            final String referenceModelsFolder,
+            final String newModelsFolder) throws IOException, InterruptedException {
+
+        // Check for differences and report in detailed manner!
+        final Process p = new ProcessBuilder(
+                "python",
+                "src/test/resources/diff_models.py",
+                referenceModelsFolder,
+                newModelsFolder
+        ).start();
+
+        // Fetch process output
+        final BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+        final StringBuilder builder = new StringBuilder();
+        String line = null;
+        while ((line = reader.readLine()) != null) {
+            builder.append(line);
+            builder.append(System.getProperty("line.separator"));
+        }
+        final String processOutput = builder.toString();
+
+        p.waitFor();
+        logger.error(processOutput);
+
+        return (p.exitValue() == 0);
+    }
+
+    /**
+     * Asserts that the two files have the same content.
+     *
+     * @param name      name of the file to compare for the assert message.
+     * @param filepath1 path to the first file
+     * @param filepath2 path to the second file
+     * @throws IOException Raised in case of failure reading the files
+     * @since 1.0.19
+     */
+    private void assertEqualFileContents(final String name,
+                                         final Path filepath1,
+                                         final Path filepath2) throws IOException {
+
+        final File file1 = new File(filepath1.toString());
+        final File file2 = new File(filepath2.toString());
+
+        assertThat(FileUtils.contentEquals(file1, file2))
+                .as(String.format("%s file comparison", name))
+                .isTrue();
+    }
+
+    /**
+     * This functional test ensures that LightGBM can read a model file and output one exactly alike the one read in.
+     * This is to ensure the new code to rewrite the model read/write layers is completely functional.
+     * The two reference models were generated initially with LightGBM's v3.0.0 code.
+     * The two generated ones will use the current code in the current locale. There should be no mismatches.
+     *
+     * @throws URISyntaxException    For invalid resource paths
+     * @throws ModelLoadingException Errors when loading the model resources
+     * @throws IOException           IO Errors opening/writing
+     * @throws InterruptedException  Thrown if the model report fails to await for the process
+     * @since 1.0.19
+     */
+    @Test
+    public void testRewriteModel() throws URISyntaxException, ModelLoadingException, IOException, InterruptedException {
```

Comment:
avoid using the prefix `test`, as it is redundant information.
Based on the Javadoc, you should name something like: `ensureRewriteModelFunctionality()`